src/main/scala/com/rbmhtechnology/calliope/serializer/kafka/PayloadFormatSerializer.scala

```diff
@@ -48,22 +51,46 @@ class PayloadFormatSerializer[A <: AnyRef] private(serializer: PayloadSerializer
 }
 
 object PayloadFormatDeserializer {
-  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
-    new PayloadFormatDeserializer[A](DelegatingStringManifestPayloadSerializer(system), payloadMapper)
 
   def apply()(implicit system: ActorSystem): PayloadFormatDeserializer[AnyRef] =
     apply[AnyRef](identity)
 
-  def create[A](payloadMapper: JFunction[AnyRef, A], system: ActorSystem): PayloadFormatDeserializer[A] =
-    apply(payloadMapper.apply)(system)
+  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
+    instance(payloadMapper, err => throw err)
+
+  def option[A](payloadMapper: PartialFunction[AnyRef, A])(implicit system: ActorSystem): PayloadFormatDeserializer[Option[A]] =
+    apply(payloadMapper.lift)
+
+  def filterError[A](payloadMapper: PartialFunction[AnyRef, A], onError: Throwable => Unit)(implicit system: ActorSystem): PayloadFormatDeserializer[Option[A]] =
```

Comment:
For `def filterError[A](payloadMapper: PartialFunction[AnyRef, A], onError: Throwable => Unit)` we could actually return a Try. 

For the proposed Java API `
  def createFilterError[A](payloadMapper: JFunction[AnyRef, JOption[A]], onError: JConsumer[Throwable], system: ActorSystem): PayloadFormatDeserializer[JOption[A]] ` I would vote for keeping it this way for two reasons: 1. keeping the dependencies for calliope to minimum 2. Java developers are familiar with an `Optional`